tests/test_bundle.py

```diff
@@ -90,6 +91,68 @@ def _test_bundle_get(self, replica: Replica):
             self.assertEqual(resp_obj.json['bundle']['files'][0]['uuid'], "ce55fd51-7833-469b-be0b-5da88ebebfcd")
             self.assertEqual(resp_obj.json['bundle']['files'][0]['version'], "2017-06-16T193604.240704Z")
 
+    @testmode.standalone
+    def test_bundle_get_paging(self):
+        bundle_uuid = "7f8c686d-a439-4376-b367-ac93fc28df43"
+        version = "2019-02-21T184000.899031Z"
+        with override_bucket_config(BucketConfig.TEST_FIXTURE):
+            handle = Config.get_blobstore_handle(Replica.aws)
+            manifest = json.loads(handle.get(
+                Replica.aws.bucket,
+                f"bundles/{bundle_uuid}.{version}"
+            ))
+            expected_files = manifest['files']
+
+        with self.subTest("Should be able to use a variety of valid paging with the same result"):
+            self._test_bundle_get_paging(Replica.aws, expected_files, 11)
+            self._test_bundle_get_paging(Replica.aws, expected_files, 33)
+            self._test_bundle_get_paging(Replica.aws, expected_files, 500)
+            self._test_bundle_get_paging(Replica.gcp, expected_files, 11)
+            self._test_bundle_get_paging(Replica.gcp, expected_files, 33)
+            self._test_bundle_get_paging(Replica.gcp, expected_files, 500)
+
+        with self.subTest("Should NOT be able to use a too-small per_page"):
+            self._test_bundle_get_paging(Replica.aws, list(), 9, codes=requests.codes.bad_request)
+            self._test_bundle_get_paging(Replica.gcp, list(), 9, codes=requests.codes.bad_request)
+
+        with self.subTest("Should NOT be able to use a too-large per_page"):
+            self._test_bundle_get_paging(Replica.aws, list(), 501, codes=requests.codes.bad_request)
+            self._test_bundle_get_paging(Replica.gcp, list(), 501, codes=requests.codes.bad_request)
+
+    def _test_bundle_get_paging(self,
+                                replica: Replica,
+                                expected_files: list,
+                                per_page: int,
+                                codes={requests.codes.ok, requests.codes.partial}):
+        replica = Replica.aws
+        bundle_uuid = "7f8c686d-a439-4376-b367-ac93fc28df43"
+        version = "2019-02-21T184000.899031Z"
+
+        url = str(UrlBuilder()
+                  .set(path="/v1/bundles/" + bundle_uuid)
+                  .add_query("replica", replica.name)
+                  .add_query("per_page", str(per_page))
+                  .add_query("version", version))
+
+        with override_bucket_config(BucketConfig.TEST_FIXTURE):
+            resp_obj = self.assertGetResponse(url, codes, headers=get_auth_header())
```

Comment:
one gap in your test coverage is that if the partial responses return `requests.codes.ok`, your test won't catch it (and vice versa).

Reply:
fixed